zip-0032.rst

```diff
@@ -239,16 +246,74 @@ in [#protocol-saplingkeycomponents]_.
     :math:`I = \mathsf{PRF^{expand}}(\mathsf{c}_{par}, [\texttt{0x12}]`:math:`||\,\mathsf{EncodeExtFVKParts}(\mathsf{ak}_{par}, \mathsf{nk}_{par}, \mathsf{ovk}_{par}, \mathsf{dk}_{par})`:math:`||\,\mathsf{I2LEOSP}_{32}(i))`.
 
 - Split :math:`I` into two 32-byte sequences, :math:`I_L` and :math:`I_R`.
-- Let :math:`I_\mathsf{ask} = \mathsf{ToScalar}(\mathsf{PRF^{expand}}(I_L, [\texttt{0x13}]))`.
-- Let :math:`I_\mathsf{nsk} = \mathsf{ToScalar}(\mathsf{PRF^{expand}}(I_L, [\texttt{0x14}]))`.
+- Let :math:`I_\mathsf{ask} = \mathsf{ToScalar^{Sapling}}(\mathsf{PRF^{expand}}(I_L, [\texttt{0x13}]))`.
+- Let :math:`I_\mathsf{nsk} = \mathsf{ToScalar^{Sapling}}(\mathsf{PRF^{expand}}(I_L, [\texttt{0x14}]))`.
 - Return:
 
-  - :math:`\mathsf{ak}_i  = [I_\mathsf{ask}]\,\mathcal{G} + \mathsf{ak}_{par}`
-  - :math:`\mathsf{nk}_i  = [I_\mathsf{nsk}]\,\mathcal{H} + \mathsf{nk}_{par}`
+  - :math:`\mathsf{ak}_i  = [I_\mathsf{ask}]\,\mathcal{G}^\mathsf{Sapling} + \mathsf{ak}_{par}`
+  - :math:`\mathsf{nk}_i  = [I_\mathsf{nsk}]\,\mathcal{H}^\mathsf{Sapling} + \mathsf{nk}_{par}`
   - :math:`\mathsf{ovk}_i = \mathsf{truncate}_{32}(\mathsf{PRF^{expand}}(I_L, [\texttt{0x15}]`:math:`||\,\mathsf{ovk}_{par}))`
   - :math:`\mathsf{dk}_i  = \mathsf{truncate}_{32}(\mathsf{PRF^{expand}}(I_L, [\texttt{0x16}]`:math:`||\,\mathsf{dk}_{par}))`
   - :math:`\mathsf{c}_i   = I_R`.
 
+Sapling internal key derivation
+-------------------------------
+
+The above derivation mechanisms produce external addresses suitable for giving out to senders.
+For each such external address, we also want to produce another address for use by wallets for
```

Comment:
nit: we really only need one internal address per spending authority, rather than separate internal diversified addresses. Furthermore, for Sapling we don't have any guarantee that the same diversifier index will produce a valid address under both the internal and external spending keys, and for that reason I don't think we should put this as "for each such external address."